### PR TITLE
Fix some user permission issue

### PR DESCRIPTION
**Pull request by alexandrebodin**
Signed-off-by: Alexandre Bodin <bodin.alex@gmail.com>

### CODE REVIEW

File: packages/strapi-plugin-users-permissions/config/policies/rateLimit.js [hunk 1]
```diff
@@ -22,7 +22,7 @@ module.exports = async (ctx, next) => {
       {
         interval: 1 * 60 * 1000,
         max: 5,
-        prefixKey: `${ctx.request.url}:${ctx.request.ip}`,
+        prefixKey: `${ctx.request.path}:${ctx.request.ip}`,
```
Comment (petersg83) [inline packages/strapi-plugin-users-permissions/config/policies/rateLimit.js:25]: Can you explain why this change was needed please?

File: packages/strapi-plugin-users-permissions/controllers/Auth.js [hunk 2]
```diff
@@ -315,29 +315,21 @@ module.exports = {
       key: 'advanced',
     });
 
+    const userInfo = _.omit(user, ['password', 'resetPasswordToken', 'role', 'provider']);
+
```
Comment (petersg83) [inline packages/strapi-plugin-users-permissions/controllers/Auth.js:319]: Is it just a refacto or there is a security fix I don't see?

File: packages/strapi-plugin-users-permissions/config/policies/rateLimit.js (hunk 1 above)
Comment (alexandrebodin) [reply packages/strapi-plugin-users-permissions/config/policies/rateLimit.js:25]: the url contains querystring params and thus you can easily bypass the rate limiting.

File: packages/strapi-plugin-users-permissions/controllers/Auth.js (hunk 2 above)
Comment (alexandrebodin) [reply packages/strapi-plugin-users-permissions/controllers/Auth.js:319]: nope just a cleanup I made because I was doing the fix there in the first place.

**Comment by nunsie** [comment]
The additional regex validation that has now been introduced at `strapi-plugin-users-permissions/controllers/validation/email-template.js` seems to have broken the ability to use any field on the USER model in the email templates other than USER.email and USER.username 😞  

Is this the intent here?

**Comment by alexandrebodin** [comment]
Hi @nunsie It is indeed the intent to avoi dany security issues. If you need to extend this You can always use the extension system  to overwrite that behaviour :)

**Comment by nunsie** [comment]
ok, thanks for the clarification @alexandrebodin !
